backends/tfhe-cuda-backend/cuda/src/integer/div_rem.cuh

```diff
@@ -679,11 +692,22 @@ __host__ void host_integer_div_rem_kb(cudaStream_t const *streams,
         int_mem_ptr->negated_quotient, nullptr, nullptr, int_mem_ptr->scp_mem_1,
         bsks, ksks, num_blocks, requested_flag, uses_carry);
 
-    host_integer_radix_negation(int_mem_ptr->sub_streams_2, gpu_indexes,
-                                gpu_count, int_mem_ptr->negated_remainder,
-                                remainder, radix_params.big_lwe_dimension,
-                                num_blocks, radix_params.message_modulus,
-                                radix_params.carry_modulus);
+    CudaRadixCiphertextFFI output_2;
+    output_2.ptr = int_mem_ptr->negated_remainder;
+    output_2.degrees = (uint64_t *)malloc(num_blocks * sizeof(uint64_t));
+    output_2.noise_levels = (uint64_t *)malloc(num_blocks * sizeof(uint64_t));
+    output_2.num_radix_blocks = num_blocks;
+    output_2.lwe_dimension = radix_params.big_lwe_dimension;
+
+    CudaRadixCiphertextFFI input_2;
+    input_2.ptr = remainder;
+    input_2.degrees = (uint64_t *)malloc(num_blocks * sizeof(uint64_t));
+    input_2.noise_levels = (uint64_t *)malloc(num_blocks * sizeof(uint64_t));
+    input_2.num_radix_blocks = num_blocks;
+    input_2.lwe_dimension = radix_params.big_lwe_dimension;
+    host_integer_radix_negation<uint64_t>(
+        int_mem_ptr->sub_streams_2, gpu_indexes, gpu_count, &output_2, &input_2,
+        radix_params.message_modulus, radix_params.carry_modulus);
```

Comment:
To avoid having to do that I introduced a legacy_host_addition entry point that is the old one and can be called in this type of situation. Some functions are called in many places :sweat_smile:  Maybe it would be better to follow the same logic here?

Reply:
Ok. We just will have to remember to remove all these `legacy_` methods at some point.